x-pack/plugins/stack_connectors/server/connector_types/stack/torq/index.ts

```diff
@@ -0,0 +1,378 @@
+      defaultMessage: 'Torq',
+    }),
+    supportedFeatureIds: [
+      AlertingConnectorFeatureId,
+      UptimeConnectorFeatureId,
+      SecurityConnectorFeatureId,
+    ],
+    validate: {
+      config: {
+        schema: schema.object(configSchemaProps),
+        customValidator: validateActionTypeConfig,
+      },
+      secrets: {
+        schema: SecretsSchema,
+      },
+      params: {
+        schema: ParamsSchema,
+      },
+    },
+    renderParameterTemplates,
+    executor: curry(executor)(),
+  };
+}
+
+function renderParameterTemplates(
+  params: ActionParamsType,
+  variables: Record<string, unknown>
+): ActionParamsType {
+  if (!params.body) return params;
+  return {
+    body: renderMustacheString(params.body, variables, 'json'),
+  };
+}
+
+function validateActionTypeConfig(
+  configObject: ActionTypeConfigType,
+  validatorServices: ValidatorServices
+) {
+  const configuredUrl = configObject.webhookIntegrationUrl;
+  let configureUrlObj: URL;
+  try {
+    configureUrlObj = new URL(configuredUrl);
+  } catch (err) {
+    throw new Error(
+      i18n.translate('xpack.stackConnectors.torq.torqConfigurationErrorNoHostname', {
+        defaultMessage: 'error configuring send to Torq action: unable to parse url: {err}',
+        values: {
+          err,
+        },
+      })
+    );
+  }
+
+  try {
+    validatorServices.configurationUtilities.ensureUriAllowed(configuredUrl);
+  } catch (allowListError) {
+    throw new Error(
+      i18n.translate('xpack.stackConnectors.torq.torqConfigurationError', {
+        defaultMessage: 'error configuring send to Torq action: {message}',
+        values: {
+          message: allowListError.message,
+        },
+      })
+    );
+  }
+
+  if (configureUrlObj.hostname !== 'hooks.torq.io' && configureUrlObj.hostname !== 'localhost') {
+    throw new Error(
+      i18n.translate('xpack.stackConnectors.torq.torqConfigurationErrorInvalidHostname', {
+        defaultMessage:
+          'error configuring send to Torq action: url must begin with https://hooks.torq.io',
+      })
+    );
+  }
+}
+
+// action executor
+export async function executor(
+  execOptions: TorqActionTypeExecutorOptions
+): Promise<ActionTypeExecutorResult<unknown>> {
+  const actionId = execOptions.actionId;
+  const { webhookIntegrationUrl } = execOptions.config;
+  const { body: data } = execOptions.params;
+  const configurationUtilities = execOptions.configurationUtilities;
+
+  const secrets: ActionTypeSecretsType = execOptions.secrets;
+  const token = secrets.token;
+
+  const axiosInstance = axios.create();
+  const result: Result<AxiosResponse, AxiosError<{ message: string }>> = await promiseResult(
+    request({
+      axios: axiosInstance,
+      url: webhookIntegrationUrl,
+      method: 'post',
+      headers: {
+        'X-Torq-Token': token || '',
+        'Content-Type': 'application/json',
+      },
+      data: JSON.parse(data || 'null'),
```

Comment:
I think we'll want to do this `JSON.parse()` in a separate step, before the request.  The reason is, the data may not be valid JSON, and we'll want a clear error message about that, logged to the console and returned in the `message`. field with an explicit error status.